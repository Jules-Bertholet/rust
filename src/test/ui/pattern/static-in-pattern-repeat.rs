#![feature(static_in_pattern)]
#![deny(unreachable_patterns)]

static ZERO: i32 = 0;
static ZERO_REF: &i32 = &0;


fn main() {
    let val = match 0 {
        ZERO => true,
        ZERO => true,
        //~^ ERROR unreachable pattern
        _ => false,
    };

    let val = match &0 {
        ZERO_REF => true,
        ZERO_REF => false,
        //~^ ERROR unreachable pattern
        _ => false,
    };
    assert_eq!(val, true);
}
