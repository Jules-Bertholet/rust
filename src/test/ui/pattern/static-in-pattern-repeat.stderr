error: unreachable pattern
  --> $DIR/static-in-pattern-repeat.rs:9:9
   |
LL |         ZERO => true,
   |         ^^^^
   |
note: the lint level is defined here
  --> $DIR/static-in-pattern-repeat.rs:2:9
   |
LL | #![deny(unreachable_patterns)]
   |         ^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

